scaffold/template/custom/helpers/index.js

```diff
@@ -0,0 +1,24 @@
+import React, { useState, useCallback, useEffect } from "react"
+
+export function useComponentLayout() {
+  const [layout, setLayout] = useState({})
+
+  const onLayout = useCallback(event => {
+    setLayout(event.nativeEvent.layout)
+  }, [])
+
+  return [layout, onLayout]
+}
+
+export const useMatchMedia = query => {
+  const mediaMatch = window.matchMedia(query)
```

Comment:
What's going to happen when this code is executed inside Android/iOS environment? Isn't `window` undefined there?
I'd maybe add some safe guards in case this is used in a code that needs to render both in web and mobile. Or perhaps document this code that is strictly for web only

Reply:
I see now that you did the check in the module side. I'd recommend then just making is obvious/document in the code what this is for and what should be done to avoid cross-platform issues